system_tests/forwarder_test.go

```diff
@@ -157,7 +158,14 @@ func createSequencer(
 
 // tmpPath returns file path with specified filename from temporary directory of the test.
 func tmpPath(t *testing.T, filename string) string {
-	return filepath.Join(t.TempDir(), filename)
+	// create a unique, maximum 10 characters-long temporary directory {name} with path as $TMPDIR/{name}
+	tmpDir, _ := os.MkdirTemp("", "")
+	t.Cleanup(func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			t.Errorf("tmpPath RemoveAll cleanup: %v", err)
```

Comment:
nit: this is a bit cryptic error log. "Failed to cleanup temp dir" may be more readable.